src/gretel_trainer/relational/core.py

```diff
@@ -32,17 +40,57 @@ class ForeignKey:
 UserFriendlyPrimaryKeyT = Optional[Union[str, List[str]]]
 
 
+class Scope(str, Enum):
+    ALL = "all"
+    PUBLIC = "public"
+    MODELABLE = "modelable"
+    EVALUATABLE = "evaluatable"
+    INVENTED = "invented"
+
+
 @dataclass
 class TableMetadata:
     primary_key: list[str]
     data: pd.DataFrame
     columns: set[str]
+    invented_table_metadata: Optional[InventedTableMetadata] = None
     safe_ancestral_seed_columns: Optional[set[str]] = None
 
 
 class RelationalData:
     def __init__(self):
         self.graph = networkx.DiGraph()
+        self.relational_jsons: dict[str, RelationalJson] = {}
+
+    def restore(self, tableset: dict[str, pd.DataFrame]) -> dict[str, pd.DataFrame]:
+        """Restores a given a tableset (presumably output from some MultiTable workflow,
```

Comment:
nit: (or with "the"?)
```suggestion
        """Restores a given tableset (presumably output from some MultiTable workflow,
```